### ISSUE TITLE
OpenId Server Hybrid Flow does not work

### ISSUE THREAD

**Issue by MichaelPetrinolis** [OrchardCMS/OrchardCore#11717]
### Describe the bug
When you enable Hybrid Flow on the server and configure a client to use Hybrid Flow, it does not work.

### Expected behavior
The OpenId Application should persist Hybrid Flow configuration and the client should be able to connect

**Comment by kevinchalet** [comment]
> @kevinchalet I could make a PR and remove the HybridFlow when configuring server (mentioning in hint that Hybrid flow will be available for applications if both basic and implicit flows are enabled)

This was the logic used in OpenIddict 1.x/2.x: for the hybrid flow to be considered allowed, it was enough to enable both the implicit and authorization code grants (the hybrid flow doesn't have a dedicated `grant_type` so it's often treated as a combination of both).

OpenIddict 3.x introduced "response type permissions" to make things much more granular and it seems I missed a few things when I updated the OpenID module to support them.

> and allow configuration of the Response Types CodeToken, CodeIdToken and CodeIdTokenToken for the application, if you agree.

Making the allowed `response_type` combinations is something we decided against to keep things simpler for the users (instead, we just have checkboxes for the different flows)

> The current code enables CodeToken and CodeIdTokenToken only for public clients. Is it because they get the access_token through the front_channel ?

Public clients can get a backchannel access token by making a token request too. That check is actually there to *prevent* confidential clients (or a malicious actor trying to downgrade the `response_type`) from getting a frontchannel access token without having to authenticate first. It's not prohibited by the spec, but it's a security design decision we made for OC :smile:
